core/generated-sources/src/test/kotlin/org/jetbrains/kotlinx/dataframe/api/ColumnsSelectionDsl.kt

```diff
@@ -0,0 +1,515 @@
+        } shouldBe df.select {
+            name.colsOf<String>()[{ "Name" in it.name() }]
+        }
+
+        df.select {
+            "name".cols { "Name" in it.name() }
+        } shouldBe df.select {
+            Person::name.cols { "Name" in it.name() }
+        }
+
+        df.select {
+            "name"[{ "Name" in it.name() }]
+        } shouldBe df.select {
+            Person::name[{ "Name" in it.name() }]
+        }
+
+        df.select {
+            pathOf("name").cols { "Name" in it.name() }
+        } shouldBe df.select {
+            pathOf("name")[{ "Name" in it.name() }]
+        }
+    }
+
+    @Test
+    fun `cols and get with column references`() {
+        df.select { all().cols(name, age) } shouldBe df.select { cols(name, age) }
+        df.select { all()[name, age] } shouldBe df.select { this[name, age] }
+
+        val firstName by column<String>()
+        val lastName by column<String>()
+        df.select {
+            name.cols(firstName, lastName)
+        } shouldBe df.select {
+            name.colsOf<String>().cols(firstName, lastName)
+        }
+
+        df.select {
+            name.cols(name.firstName, name.lastName)
+        } shouldBe df.select {
+            name.colsOf<String>().cols(name.firstName, name.lastName)
+        }.also { it.print() }
+
+        df.select {
+//            name[name.firstName, name.lastName]
+            name.cols(name.firstName, name.lastName)
+        } shouldBe df.select {
+            name.colsOf<String>()[name.firstName, name.lastName]
+        }
+
+        df.select {
+            "name".cols(name.firstName, name.lastName)
+        } shouldBe df.select {
+            Person::name.cols(name.firstName, name.lastName)
+        }
+
+        df.select {
+            "name"[name.firstName, name.lastName]
+        } shouldBe df.select {
+            Person::name[name.firstName, name.lastName]
+        }
+
+        df.select {
+            pathOf("name").cols(name.firstName, name.lastName)
+        } shouldBe df.select {
+            pathOf("name")[name.firstName, name.lastName]
+        }
+    }
+
+    @Test
+    fun `cols and get with column names`() {
+        df.select { all().cols("name", "age") } shouldBe df.select { cols("name", "age") }
+        df.select { all()["name", "age"] } shouldBe df.select { this["name", "age"] }
+
+        df.select {
+            name.cols("firstName", "lastName")
+        } shouldBe df.select {
+            name.colsOf<String>().cols("firstName", "lastName")
+        }
+
+        df.select {
+//            name["firstName", "lastName"]
+            name.cols("firstName", "lastName")
+        } shouldBe df.select {
+            name.colsOf<String>()["firstName", "lastName"]
+        }
+
+        df.select {
+            "name".cols("firstName", "lastName")
+        } shouldBe df.select {
+            Person::name.cols("firstName", "lastName")
+        }
+
+        df.select {
+            "name"["firstName", "lastName"]
```

Comment:
it still looks more like "pathOf" for me than slicing

Reply:
pathOf would be `"name"["firstName"]["lastName"]`. I actually think this is kinda neat. It allows you to select multiple columns way down in a path: `"some"["path"]["to"]["multiple", "columns"]`

Reply:
I think it takes a bit more effort to "parse" this expression and recognize the difference and to understand that the result is multiple columns.
Here the difference is more obvious
`"some"["path"]["to"].cols("multiple", "columns")`

Reply:
Yeah for sure :) But again for completeness, should we remove it?
`pathOf("path", "to")["a", "b"]` doesn't look that wrong

Reply:
Maybe? I don't know, let's check how it looks in completion. How discoverable is this API, how many options there are